gen/calc.py

```diff
@@ -224,6 +224,17 @@ def validate_host_list(host_list):
         return host_list
     assert not any(azure_format_check), "Azure static master list and IP based static master list cannot be mixed. Use "
     "either all Azure IP references or IPv4 addresses."
+
```

Comment:
The duplicate check happens on the list of hosts after json decode, so duplicate IPs with different ports wouldn't break the assertion as written.

Re: separating the function, I see now that I did break the norm by adding a default argument, will address